### DISCUSSION TITLE
Locate H2 DB in persistent mode within a docker container

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#351] [General]
I am running a keyfactor/ejbca-ce on ubuntu 22 with h2 DB but not able to locate the H2 db in container file system so can any one guide me where does it store when docker is shut down gracefully.

See my docker image inspect details.

See my wildfly standalone subsystem setting below:

 <subsystem xmlns="urn:jboss:domain:datasources:7.0">
            <datasources>
                <datasource jndi-name="java:/AppDS" pool-name="appds" use-ccm="true" statistics-enabled="${wildfly.datasources.statistics-enabled:${wildfly.statistics-enabled:false}}">
                    <connection-url>${env.DATABASE_JDBC_URL:jdbc:h2:mem:ejbcadb;DB_CLOSE_DELAY=-1}</connection-url>
                    <driver-class>${env.DATABASE_JDBC_DRIVER_CLASS:org.h2.Driver}</driver-class>
                    <driver>${env.DATABASE_JDBC_DRIVER:h2}</driver>
                    <transaction-isolation>${env.DATABASE_JDBC_ISOLATION:TRANSACTION_READ_COMMITTED}</transaction-isolation>
                    <pool>
                        <min-pool-size>10</min-pool-size>
                        <max-pool-size>150</max-pool-size>
                        <prefill>true</prefill>
                    </pool>
                    <security>
                        <user-name>${env.DATABASE_USER:sa}</user-name>
                        <password>${env.DATABASE_PASSWORD:sa}</password>
                    </security>
                    <validation>
                        <check-valid-connection-sql>${env.DATABASE_CONNECTION_CHECK:SELECT 1;}</check-valid-connection-sql>
                        <validate-on-match>false</validate-on-match>
                        <background-validation>true</background-validation>
                        <background-validation-millis>30000</background-validation-millis>
                    </validation>
                    <timeout>
                        <idle-timeout-minutes>3</idle-timeout-minutes>
                    </timeout>
                    <statement>
                        <prepared-statement-cache-size>${env.DATABASE_PREPARED_STATEMENT_CACHE_SIZE:50}</prepared-statement-cache-size>
                        <share-prepared-statements>${env.DATABASE_SHARE_PREPARED_STATEMENTS:true}</share-prepared-statements>
                    </statement>
                </datasource>
                <drivers>
                    <driver name="h2" module="com.h2database.h2">
                        <xa-datasource-class>org.h2.jdbcx.JdbcDataSource</xa-datasource-class>
                    </driver>
                </drivers>
            </datasources>
        </subsystem>
[docker_inspect-post.txt](https://github.com/Keyfactor/ejbca-ce/files/12651609/docker_inspect-post.txt)

**Comment by primetomas** [comment]
You can see in the documentation: https://hub.docker.com/r/keyfactor/ejbca-ce

jdbc:h2:/mnt/persistent/ejbcadb;DB_CLOSE_DELAY=-1 will use the bundled H2 database that is persisted when the container is gracefully stopped. This is the default value.
